File: leetcode/1. Two Sum/solution.scala
object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    var response:Array[Int] = Array.empty[Int]
    var required = Map[Int, Int]()

    for( (num, index) <- nums.zipWithIndex) {
 
        var required_num = target - num
        if required.contains(required_num) && response.isEmpty then
            response = Array[Int](required.getOrElse(required_num, -1), index)
 
        required = required + (num -> index)
    }
    response
  }

}

File: leetcode/1. Two Sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        for(int i=0; i<nums.length-1; i++){
            for(int j =i+1; j<nums.length; j++){
                if(nums[i]+nums[j] == target){
                    int[] response = {i, j};

                    return response;
                }
            }
        }
        return null;
    }
}